va/validation-authority.go

```diff
@@ -54,6 +54,26 @@ type verificationRequestEvent struct {
 
 // Validation methods
 
+// parseDNSError checks the error returned from Lookup... methods and tests if the error
+// was an underlying net.OpError or an error caused by resolver returning SERVFAIL or other
+// invalid Rcodes.
+func (va ValidationAuthorityImpl) parseDNSError(err error, challenge core.Challenge) core.Challenge {
+	challenge.Error = &core.ProblemDetails{Type: core.ServerInternalProblem}
```

Comment:
Let's not use `ServerInternalProblem` since it's so commonly intercepted, and this isn't a situation of Boulder losing its mind.
`ConnectionProblem` or `UnknownHostProblem` seem like perfectly fine choices.